### ISSUE TITLE
`@ts-nocheck` not working

### ISSUE THREAD

**Issue by kakariko-village** [ezolenko/rollup-plugin-typescript2#223]
#### What happens and why it is wrong

Hi, i put `// @ts-nocheck` on the top of my tsx file but still got `semantic error TS2554` when run  `rollup -c`

#### Versions
- typescript: 3.8.3
- rollup: 2.3.2
- rollup-plugin-typescript2: 0.27.0

#### rollup.config.js

```js
import typescript from 'rollup-plugin-typescript2';
import pkg from './package.json';

export default [
    {
        input: 'src/index.tsx',
        output: [
            {
                file: 'dist/index.js',
                format: 'esm',
                banner: '/* eslint-disable */',
            },
            { file: pkg.main, format: 'cjs' },
            { file: pkg.module, format: 'esm' },
        ],
        plugins: [
            typescript(),
        ],
        onwarn: function(warning) {
            // Skip certain warnings
            
            if ( warning.code === 'THIS_IS_UNDEFINED' ) { return; }
            
            console.warn( warning.message );
        },
        external: Object.keys(pkg.peerDependencies || {}),
    },
];
```

#### tsconfig.json

```json5
{
  "compilerOptions": {
    "target": "esnext",
    "module": "esnext",
    "moduleResolution": "node",
    "importHelpers": true,
    "jsx": "react",
    "esModuleInterop": true,
    "sourceMap": true,
    "baseUrl": ".",
    "strict": false,
    "paths": {
      "@/*": ["src/*"]
    },
    "declaration": true,
    "outDir": "./dist",
    "allowSyntheticDefaultImports": true,
    "resolveJsonModule": true,
    "strictNullChecks": false,
    "removeComments": true,
    "noLib": false,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "noEmitOnError": false
  }
}
```

#### package.json

```json
{
  "name": "test",
  "version": "0.1",
  "main": "dist/index.js",
  "module": "dist/index.esm.js",
  "scripts": {
    "build": "rollup -c"
  },
  "dependencies": {
    "antd": "3.23.6",
    "father": "^2.29.2",
    "hammerjs": "^2.0.8",
    "react": "^16.8.6",
    "react-dom": "^16.8.6",
    "react-live": "^2.2.2",
    "react-perfect-scrollbar": "^1.5.3",
    "react-signature-canvas": "^1.0.3",
    "react-simple-keyboard": "^2.0.5",
    "react-slick": "^0.25.2",
    "react-sticky-box": "^0.9.2",
    "react-syntax-highlighter": "^11.0.2",
    "react-window-size": "^1.2.2",
    "recharts": "^2.0.0-beta.1",
    "rxjs": "^6.5.3",
    "slick-carousel": "^1.8.1",
    "typescript": "3.6.4",
    "umi-plugin-locale": "^2.11.2"
  },
  "devDependencies": {
    "@types/hammerjs": "^2.0.36",
    "@types/jest": "^23.3.12",
    "@types/react": "^16.7.18",
    "@types/react-dom": "^16.0.11",
    "@types/react-test-renderer": "^16.0.3",
    "babel-eslint": "^9.0.0",
    "eslint": "^5.4.0",
    "eslint-config-umi": "^1.4.0",
    "eslint-plugin-flowtype": "^2.50.0",
    "eslint-plugin-import": "^2.14.0",
    "eslint-plugin-jsx-a11y": "^5.1.1",
    "eslint-plugin-react": "^7.11.1",
    "husky": "^0.14.3",
    "lint-staged": "^7.2.2",
    "react-test-renderer": "^16.7.0",
    "rollup": "^2.3.2",
    "rollup-plugin-delete": "^1.2.0",
    "rollup-plugin-typescript2": "^0.27.0",
    "tslint": "^5.12.0",
    "tslint-eslint-rules": "^5.4.0",
    "tslint-react": "^3.6.0",
    "umi": "^2.9.0",
    "umi-plugin-react": "^1.8.0",
    "umi-types": "^0.3.0"
  },
  "publishConfig": {
    "registry": "https://registry.npmjs.org/"
  },
  "sideEffects": [
    "dist/*",
    "*.less"
  ],
  "typings": "dist/src/index.d.ts"
}
```

#### plugin output with verbosity 3

```
[!] (plugin rpt2) Error: /index.tsx(100,11): semantic error TS2554: Expected 1 arguments, but got 0.
```

**Comment by ezolenko** [comment]
Does it work correctly with tsc? Make sure the plugin picks up correct typescript version (it prints the version it uses during the build), this option was only implemented since 3.7 for ts files.

**Comment by agilgur5** [comment]
I created a minimal reproduction [here](https://stackblitz.com/edit/rpt2-issue-223?file=src/index.ts) and was unable to reproduce this error.

Neither `tsc` nor rpt2 give a `TS2554` error in their latest versions (`4.7.3` for TS, `0.32.1` for rpt2) when `// @ts-nocheck` is set at the top of the file.

Looking at your `package.json`, it looks like you're using an older version of TS in your project than the one you reported:

> * typescript: 3.8.3

vs.

> ```json
>     "typescript": "3.6.4",
> ```

Note that it's also pinned in the `package.json`, so it's hard set to that specific version, `3.6.4`, in your project.

As ezolenko pointed out above, `// @ts-nocheck` only began to be supported [in TS 3.7](https://devblogs.microsoft.com/typescript/announcing-typescript-3-7/#ts-nocheck-in-typescript-files) for `.ts` files.
So `// @ts-nocheck` on a `.tsx` file would not work with either `tsc` or rpt2 when using TS 3.6 as you were. Please upgrade TS to a compatible version if you're hitting this issue.
